framework/markdown.go

```diff
@@ -0,0 +1,116 @@
+/*
+ Copyright 2025 The OSCAL Compass Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package framework
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"strings"
+	"text/template"
+
+	oscalTypes "github.com/defenseunicorns/go-oscal/src/types/oscal-1-1-2"
+)
+
+//go:embed template/*.md
+var embeddedResources embed.FS
+
+type TemplateValues struct {
+	Catalog           string
+	Component         string
+	AssessmentResults oscalTypes.AssessmentResults
+}
+
+// Get the catalog title as the tamplate.md catalog info
+func getCatalogTitle(catalog oscalTypes.Catalog) string {
+	if catalog.Metadata.Title != "" {
+		return catalog.Metadata.Title
+	} else {
+		fmt.Println("Error getting catalog title")
```

Comment:
Should this return an error? If the goal is not to halt execution, maybe the logger could be passed and this could be logged as a warning?

Reply:
Thanks @qduanmu. I agree and will resolve this comment.